client/src/stellar_error.rs

```diff
@@ -32,53 +32,43 @@ pub enum ErrorType {
     UnknownError,
 }
 
-impl<'de> Deserialize<'de> for ErrorType {
+impl<'de> Deserialize<'de> for Kind {
     fn deserialize<D>(d: D) -> Result<Self, D::Error>
     where
         D: Deserializer<'de>,
     {
-        let type_enum: ErrorType = StellarError::error_type_from_str(&String::deserialize(d)?[..]);
-        Ok(type_enum)
+        let kind = String::deserialize(d)?;
+        Kind::from_str(&kind).map_err(|_| de::Error::custom("Error decoding kind"))
     }
 }
 
-impl StellarError {
-    /// Creates a new stellar error.  If the error type is not recognized an error is returned.
-    pub fn new(
-        error_type: &str,
-        title: String,
-        status: u16,
-        detail: String,
-        instance: String,
-    ) -> Result<StellarError, String> {
-        let type_enum: ErrorType = StellarError::error_type_from_str(error_type);
-        Ok(StellarError {
-            error_type: type_enum,
-            title: title,
-            status: status,
-            detail: detail,
-            instance: instance,
+impl FromStr for Kind {
+    type Err = super::Error;
+
+    fn from_str(s: &str) -> Result<Self, super::Error> {
```

Comment:
Is there an added benefit to keeping this outside of the deserializer?  I see you use this in a test, but could also use the same deserialize pattern we use in other resource files.  Curious to hear your thoughts.

Reply:
Well, it was captured outside the deserializer before. It was in the `error_from_str` method and we exposed a `new` function along with it for testing.

I haven't actually added to the public interface so much as moved it to a more specific Trait that's used in rust.

I do think it makes sense to `impl FromStr` when we can :)